### PR TITLE
Allow ssl_opts to be overridden when updating

### PR DESCRIPTION
**Pull request by danielspofford**
Why
---

- Flexibility for consumers.

How
---

- Accept `opts` in `NervesHubLinkHTTP.update` and pass them along.

### CODE REVIEW

File: lib/nerves_hub_link_http.ex [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
+      {:ok, url} when is_binary(url) -> apply_update(url)
+      {:error, _} = err -> err
     end
   end
 
+  @doc """
+  Enter a receive loop handling messages from `NervesHubLinkHTTP.HTTPFwupStream`.
+  """
+  @spec update_receive() :: no_return()
   def update_receive() do
```
Comment (jjcarstens) [inline lib/nerves_hub_link_http.ex]: I feel like maybe this should just be private? Is there any case we need to allow the caller access to this?

Comment (danielspofford) [reply lib/nerves_hub_link_http.ex]: I have never needed to call this manually myself. Happy to go either way here as to not hold up the other changes.

Comment (jjcarstens) [reply lib/nerves_hub_link_http.ex]: Sweet. I feel like we can remove the doc/spec here and just make it private if you're good with making that change (even though its not _really_ what you were looking to change)

**Comment by danielspofford** [comment]
@jjcarstens I made `update_receive/0` private 👍

**Review by jjcarstens — APPROVED** [review]

**Pull request merged**